packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -115,6 +121,58 @@ class DatePickerInput extends React.Component {
     }
   }
 
+  static getDerivedStateFromProps(props, state) {
+    // Formats the selected date into a consistent value and sets input state.
+    if (props.value !== state.prevDate) {
+      const format = DateUtil.getDateFormat(props.intl.formatMessage({ id: 'Terra.datePicker.dateFormatOrder' }));
```

Comment:
Why can't we use the date format already supplied for the locale? https://github.com/cerner/terra-framework/blob/master/packages/terra-date-picker/translations/nl.json#L2

If anything I could see adding a data format separator so we can update the below to substring parsing